### ISSUE TITLE
Ability to limit data sources to a bounding box

### ISSUE THREAD

**Issue by bcamper** [tangrams/tangram#366]
For tiled data sources (vector or raster), it would be useful in some cases (for more localized data sets) to limit tiles to a specific bounding box, outside of which no tiles would be requested/displayed. 

As a more advanced option, we could consider specifying a "fallback" tile that is always used outside of the bounding box.

**Comment by dmvaldman** [comment]
Could this feature also be used for offline maps? It would be interesting to have an estimate of how much data is in, say, a large metropolitan area like Manhattan and whether this could fit in localstorage, or IndexDB.

**Comment by bcamper** [comment]
Yes! :)

**Comment by bcamper** [comment]
@thisisaaronland experimented with local storage of tiles in tangram several months ago

**Comment by thisisaaronland** [comment]
this is as far as I got: 

https://github.com/thisisaaronland/tangram/blob/localforage/src/utils/utils.js#L130-L151

Note this is probably several (Tangram) releases out of date...

**Comment by dmvaldman** [comment]
`localforage`?

**Comment by thisisaaronland** [comment]
https://mozilla.github.io/localForage/

**Comment by bcamper** [comment]
WIP in https://github.com/tangrams/tangram/compare/source-bounds

**Comment by hasan9206** [comment]
I was about to submit an issue asking if this was possible, I found that my map was rendering slower when I added a new data source, this new source contains the buildings footprints (which I then extrude) only in a few areas of the map.

When I did some testing I realised that the source was being called everywhere on the map, a request was being sent and it kept getting a 'tile not found error' (since the building data is pretty limited, where there are no buildings, there are no tiles) which was slowing down my rendering speed significantly.

So I thought about trying to limit my data source to a particular region ( only where it was being called) but I didn't really know how I should do this in Tangram.
